test/recurring/constants.js

```diff
@@ -1,46 +1,131 @@
+const REF_ID = 'ref-123';
+const BUSINESS_ID = 'fake-BID';
+const CUSTOMER_ID = 'cust-123';
+const CURRENCY = 'IDR';
 const EXT_ID = '123';
 const PAYER_EMAIL = 'dummy@email.co';
 const DESCRIPTION = 'Payment for something';
 const AMOUNT = 10000;
 const INTERVAL = 'MONTH';
 const INTERVAL_COUNT = 1;
 const PAYMENT_ID = '5e0577bdf4d38b20d542009b';
+const SCHEDULE_ID = '5e0577bdf4d38b20d542009b';
+const ANCHOR_DATE = new Date().toISOString();
 const START_DATE = '2020-01-31T17:00:00.000Z';
 const CUSTOMER = {
   given_names: 'stan',
   email: 'dummy@email.co',
   mobile_number: '',
   address: '',
 };
+const PAYMENT_METHOD_ID = 'pm-123';
+const UPDATED_AMOUNT = 20000;
 
-const PAYMENT_DETAILS = {
-  id: PAYMENT_ID,
-  external_id: EXT_ID,
-  payer_email: PAYER_EMAIL,
-  description: DESCRIPTION,
-  amount: AMOUNT,
+const SCHEDULE_DETAILS = {
+  reference_id: REF_ID,
   interval: INTERVAL,
+  business_id: BUSINESS_ID,
   interval_count: INTERVAL_COUNT,
-  start_date: START_DATE,
-  customer: CUSTOMER,
+  total_recurrence: INTERVAL_COUNT + 2,
+  anchor_date: ANCHOR_DATE,
+  retry_interval: INTERVAL,
+  retry_interval_count: INTERVAL_COUNT,
+  total_retry: INTERVAL_COUNT,
+  failed_attempt_notifications: [INTERVAL_COUNT],
 };
 
-const UPDATED_AMOUNT = 20000;
-const UPDATED_PAYMENT_DETAILS = Object.assign({}, PAYMENT_DETAILS, {
+const UPDATED_SCHEDULE_DETAILS = Object.assign({}, SCHEDULE_DETAILS, {
+  interval_count: INTERVAL_COUNT + 1,
+});
+
+const PLAN_ID = 'repl-123';
+
+const PLAN_DETAILS = {
+  id: PLAN_ID,
+  reference_id: REF_ID,
+  business_id: BUSINESS_ID,
+  customer_id: CUSTOMER_ID,
+  recurring_action: 'PAYMENT',
+  recurring_cycle_count: 0,
+  currency: 'IDR',
+  amount: 1000,
+  status: 'SUCCEEDED',
+  created: new Date().toISOString(),
+  updated: new Date().toISOString(),
+  schedule_id: SCHEDULE_ID,
+  payment_methods: [],
+  immediate_action_type: 'FULL_AMOUNT',
+  notification_config: {
+    recurring_created: ['email'],
+    recurring_succeeded: ['sms'],
+    recurring_failed: ['whatsapp'],
+  },
+  failed_cycle_action: 'RESUME',
+  metadata: { foo: 'bar' },
+  description: 'baz',
+};
+
+const UPDATED_PLAN_DETAILS = Object.assign({}, PLAN_DETAILS, {
   amount: UPDATED_AMOUNT,
 });
 
+const CYCLE_ID = 'recy-1234';
+const SCHEDULED_TIMESTAMP = new Date('2022-03-07T16:08:58.697Z').toISOString();
```

Comment:
Is this fine to be hardcoded? Or should it be dynamic?

Reply:
i think for tests we have a lot of things hard coded. Do you see any issue with that?

Reply:
My only concern is, will that affect the logic in terms of the unit test failure occurence, the next author who will create the PR might need to look for the modules owner and seek for clarity.

Just to reconfirm if the fixed date won't cause any unit test failure in future.

Reply:
i see, i don't think there will be an issue, because we `nock` the request. But i changed it to be dynamic for more safety and to avoid confusions